tests/foreman/cli/test_activationkey.py

```diff
@@ -49,1578 +49,1577 @@
-        ContentView.publish({'id': new_cv['id']})
-        cvv = ContentView.info({'id': new_cv['id']})['versions'][0]
-        ContentView.version_promote({'id': cvv['id'], 'to-lifecycle-environment-id': env['id']})
-        ActivationKey.update(
+        activation_key = ActivationKey.info({'id': activation_key['id']})
+        assert activation_key['host-collections'][0]['name'] == host_col['name']
+    finally:
+        Defaults.delete({'param-name': 'organization_id'})
+
+
+@pytest.mark.run_in_one_thread
+@skip_if_not_set('fake_manifest')
+@pytest.mark.tier3
+def test_positive_add_redhat_product(module_org):
+    """Test that RH product can be associated to Activation Keys
+
+    :id: 7b15de8e-edde-41aa-937b-ad6aa529891a
+
+    :expectedresults: RH products are successfully associated to Activation
+        key
+
+    :CaseLevel: System
+    """
+    org = make_org()
+    # Using CDN as we need this repo to be RH one no matter are we in
+    # downstream or cdn
+    result = setup_org_for_a_rh_repo(
+        {
+            'product': PRDS['rhel'],
+            'repository-set': REPOSET['rhst7'],
+            'repository': REPOS['rhst7']['name'],
+            'organization-id': org['id'],
+        },
+        force_use_cdn=True,
+    )
+    content = ActivationKey.product_content(
+        {'id': result['activationkey-id'], 'organization-id': org['id']}
+    )
+    assert content[0]['name'] == REPOSET['rhst7']
+
+
+@pytest.mark.tier3
+@pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
+def test_positive_add_custom_product(module_org):
+    """Test that custom product can be associated to Activation Keys
+
+    :id: 96ace967-e165-4069-8ff7-f54c4c822de0
+
+    :expectedresults: Custom products are successfully associated to
+        Activation key
+
+    :CaseLevel: System
+
+    :BZ: 1426386
+    """
+    result = setup_org_for_a_custom_repo(
+        {'url': FAKE_0_YUM_REPO, 'organization-id': module_org.id}
+    )
+    repo = Repository.info({'id': result['repository-id']})
+    content = ActivationKey.product_content(
+        {'id': result['activationkey-id'], 'organization-id': module_org.id}
+    )
+    assert content[0]['name'] == repo['name']
+
+
+@pytest.mark.run_in_one_thread
+@skip_if_not_set('fake_manifest')
+@pytest.mark.tier3
+@pytest.mark.upgrade
+@pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
+def test_positive_add_redhat_and_custom_products(module_org):
+    """Test if RH/Custom product can be associated to Activation key
+
+    :id: 74c77426-18f5-4abb-bca9-a2135f7fcc1f
+
+    :Steps:
+
+        1. Create Activation key
+        2. Associate RH product(s) to Activation Key
+        3. Associate custom product(s) to Activation Key
+
+    :expectedresults: RH/Custom product is successfully associated to
+        Activation key
+
+    :CaseLevel: System
+
+    :BZ: 1426386
+    """
+    org = make_org()
```

Comment:
Is this necessary when you can call `module_org`?

Reply:
certain tests fail when they run back to back using the same fixture, I had to create a separate org for these tests